### PR TITLE
Xsrust/mvp

### PR DESCRIPTION
**Pull request by xsrust**
Addresses #242, with implementation as detailed in the ticket

Note: this code will require a db migration to add the 'active' flag to the roles table

### CODE REVIEW

**Comment by xsrust** [comment]
Forgot to check out a new branch before committing changes for #492.  

So now this PR includes the work for public DMP export with multiple phases

File: app/controllers/roles_controller.rb [hunk 1]
```diff
@@ -66,6 +66,21 @@ def destroy
     redirect_to controller: 'plans', action: 'share', id: @role.plan.id
   end
 
+  def archive
```
Comment (vyruss) [inline app/controllers/roles_controller.rb]: I'm not 100% on the `archive` action name for this. Perhaps `deactivate` is a bit clearer on what it does - "remove" is certainly not accurate (even though it eventually refers to "remove plan" action)

File: app/policies/role_policy.rb [hunk 2]
```diff
@@ -19,4 +19,8 @@ def update?
   def destroy?
     @role.plan.owned_by?(@user.id)
   end
+
+  def archive?
```
Comment (vyruss) [inline app/policies/role_policy.rb]: Again not 100% sold on the name

File: app/views/plans/index.html.erb [hunk 3]
```diff
@@ -7,24 +7,24 @@
 
 <!-- if the user has projects -->
 <p>
-<% if @plans.count > 0 %>
+<% if @plans.length > 0 %>
```
Comment (vyruss) [inline app/views/plans/index.html.erb]: good change.

File: config/routes.rb [hunk 4]
```diff
@@ -224,7 +224,11 @@
       end
     end
 
-    resources :roles, only: [:create, :update, :destroy]
+    resources :roles, only: [:create, :update, :destroy] do
+      member do
+        put :archive
```
Comment (vyruss) [inline config/routes.rb]: Same comment about action name.

File: app/policies/plan_policy.rb [hunk 5]
```diff
@@ -9,47 +9,47 @@ def initialize(user, plan)
   end
 
   def show?
-    @plan.readable_by?(@user.id)
+    @plan.readable_by?(@user.id) && Role.find_by(user_id: @user.id, plan_id: @plan.id).active
```
Comment (briri) [inline app/policies/plan_policy.rb]: It makes sense to keep this active check out of the `readable_by?` method. Is the check necessary though? These plans are filtered out of the user's dashboard, so the likelihood of them hitting these policy checks is slim. Even if they did via a bookmark or something, they should technically be able to access them.

This is maybe a discussion for a separate ticket though. I could see use cases where we would send them to these endpoints to view the plan and restore/unhide them if we decide to offer that option.

File: app/models/role.rb [hunk 6]
```diff
@@ -37,6 +38,10 @@ def access_level
     end
   end
 
+  def set_defaults
+    self.active = true if self.new_record?
```
Comment (xsrust) [inline app/models/role.rb]: Thanks.  stole it from some guy's blog on best practices for setting defaults in rails

File: app/policies/plan_policy.rb (hunk 5 above)
Comment (xsrust) [inline app/policies/plan_policy.rb]: The check is probably not needed, but If a user removes themself from a plan, and other users cant see that they can access the plan, I like ensuring that behavior in code.

**Comment by xsrust** [comment]
Just realized while reading over your comments that I didn't filter the 'share plan' page to remove the roles which are not active.  I'll add that, make the recommended changes, and update the PR.

**Comment by briri** [comment]
looks good @xsrust

**Pull request merged**